cli/src/native.rs

```diff
@@ -96,12 +97,15 @@ async fn fetch_bridge_multiaddrs(ws_url: &str) -> Result<Vec<Multiaddr>> {
     let addrs = bridge_info
         .addrs
         .into_iter()
-        .filter(|ma| ma.protocol_stack().any(|protocol| protocol == "tcp"))
-        .map(|mut ma| {
-            if !ma.protocol_stack().any(|protocol| protocol == "p2p") {
-                ma.push(Protocol::P2p(bridge_info.id.into()))
+        .filter_map(|mut ma| {
+            if ma.protocol_stack().any(|protocol| protocol == "tcp") {
+                if !ma.protocol_stack().any(|protocol| protocol == "p2p") {
+                    ma.push(Protocol::P2p(bridge_info.id.into()));
+                }
+                Some(ma)
+            } else {
+                None
```

Comment:
we prefer the chain of `.filter()` and `.map()` here. the `.filter_map` is only clearer when you are already working with `Option`s or if you use the `?` operator. Here that is not a case and just introduces another level of nesting.

Reply:
Thank you for your feedback. I have reverted the code to use a chain of `.filter()` and `.map()` methods. Indeed, I understand now that in this case, using `.filter_map()` introduces an unnecessary level of nesting and that separating them is more appropriate for code readability.